pkg/byteorder/byteorder.go

```diff
@@ -0,0 +1,23 @@
+package byteorder
+
+import (
+	"encoding/binary"
+	"bytes"
+)
+
+// byteArray creates a newly allocated buffer and writes the data into it.
+func byteArray(data interface{}) []byte {
+	buf := new(bytes.Buffer)
+	binary.Write(buf, nativeOrder, data)
+	return buf.Bytes()
+}
+
+// Swab16 swaps the endianness of n.
+func Swab16(n uint16) uint16 {
```

Comment:
Instead of this, it would be nice to have `ntohs` and `htons` equivalents instead to document where we are converting from and to network byte order.